test/rendering/cases/layer-vector-decluttering-extended/main.js

```diff
@@ -0,0 +1,159 @@
+import CircleStyle from '../../../../src/ol/style/Circle.js';
+import Feature from '../../../../src/ol/Feature.js';
+import Fill from '../../../../src/ol/style/Fill.js';
+import Map from '../../../../src/ol/Map.js';
+import Point from '../../../../src/ol/geom/Point.js';
+import Stroke from '../../../../src/ol/style/Stroke.js';
+import Style from '../../../../src/ol/style/Style.js';
+import Text from '../../../../src/ol/style/Text.js';
+import VectorLayer from '../../../../src/ol/layer/Vector.js';
+import VectorSource from '../../../../src/ol/source/Vector.js';
+import View from '../../../../src/ol/View.js';
+
+const center = [1825927.7316762917, 6143091.089223046];
+const map = new Map({
+  pixelRatio: 1,
+  target: 'map',
+  view: new View({
+    center: center,
+    zoom: 12.7,
+  }),
+});
+
+const source1 = new VectorSource();
+source1.addFeatures([
+  new Feature({
+    geometry: new Point([center[0] + 1000, center[1] + 1000 - 200]),
+    text: 'center',
+  }),
+  new Feature({
+    geometry: new Point([center[0] + 1000 - 540, center[1] + 1000]),
+    text: 'west',
+  }),
+  new Feature({
+    geometry: new Point([center[0] + 1000 + 540, center[1] + 1000 - 400]),
+    text: 'east',
+  }),
+]);
+// circles are always drawn, but serve as obstacles,
+// texts are decluttered against each other and the circles
+map.addLayer(
+  new VectorLayer({
+    declutter: true,
+    source: source1,
+    zIndex: 3,
+    style: function (feature) {
+      return new Style({
+        image: new CircleStyle({
+          radius: 10,
+          stroke: new Stroke({
+            color: 'red',
+            width: 8,
+          }),
+          declutterMode: 'obstacle',
+        }),
+        text: new Text({
+          text: feature.get('text'),
+          font: 'italic bold 18px Ubuntu',
+          textBaseline: 'bottom',
+          offsetY: -15,
+        }),
+      });
+    },
+  })
+);
+
+const source2 = new VectorSource();
+source2.addFeatures([
+  new Feature({
+    geometry: new Point([center[0], center[1]]),
+    text: 'center',
+    zIndex: 2,
```

Comment:
You introduced a typo: "zindex" instead of "zIndex". Thus the red circles were not used for the orange labels, as those were on top of  the red layer.

But this highlighted that I somehow did not commit the continue in the executor: the obstacles are used in the declutter executor to be added to the declutter tree, but rendered in the normal executor - so that they are behind the labels (in the test they were also rendered in the declutter executor thus being drawn on top of the label).

Fixed the executor and updated the test with your version + fix + additional blue circle/text to test
* that objects (red) cannot be obstacles for layers with a higher z index (blue)
* that the obstacles (red) are drawn behind layers with a higher z index (blue)